jaeger-core/src/main/java/com/uber/jaeger/propagation/B3TextMapCodec.java

```diff
@@ -47,44 +49,52 @@
 
   @Override
   public void inject(SpanContext spanContext, TextMap carrier) {
-    carrier.put(TRACE_ID_NAME, HexCodec.toLowerHex(spanContext.getTraceId()));
-    if (spanContext.getParentId() != 0L) { // Conventionally, parent id == 0 means the root span
-      carrier.put(PARENT_SPAN_ID_NAME, HexCodec.toLowerHex(spanContext.getParentId()));
-    }
-    carrier.put(SPAN_ID_NAME, HexCodec.toLowerHex(spanContext.getSpanId()));
-    carrier.put(SAMPLED_NAME, spanContext.isSampled() ? "1" : "0");
-    if (spanContext.isDebug()) {
-      carrier.put(FLAGS_NAME, "1");
+    try {
+      carrier.put(TRACE_ID_NAME, HexCodec.toLowerHex(spanContext.getTraceId()));
+      if (spanContext.getParentId() != 0L) { // Conventionally, parent id == 0 means the root span
+        carrier.put(PARENT_SPAN_ID_NAME, HexCodec.toLowerHex(spanContext.getParentId()));
+      }
+      carrier.put(SPAN_ID_NAME, HexCodec.toLowerHex(spanContext.getSpanId()));
+      carrier.put(SAMPLED_NAME, spanContext.isSampled() ? "1" : "0");
+      if (spanContext.isDebug()) {
+        carrier.put(FLAGS_NAME, "1");
+      }
+    } catch (RuntimeException ex) {
+      log.error("Error when injecting SpanContext into carrier. Handling gracefully.", ex);
```

Comment:
I like the idea of a try catch, I am slightly concerned about what happens when carrier.put somehow results in corrupt data being injected. For e.g., let's take line 60, in today's world, if an exception is thrown, the thread dies. 
After this change, the data is injected (potentially corrupted), and extract (on potentially a different application with an older version of Jaeger) would have to deal with it. 

As such, I don't think it's that big of a concern looking at the TextMap code.

Reply:
There is no easy way to prevent partial injection.
* Tracing data shouldn't affect the business logic so we're safe here.
* I agree that it can possibly cause problems on the other side of the wire if it's not "resilient" enough, but still, it is improvement over dying right away IMHO :)